perun-core/src/main/java/cz/metacentrum/perun/core/blImpl/AuthzResolverBlImpl.java

```diff
@@ -197,37 +197,46 @@ public static boolean isAuthorizedForAttribute(PerunSession sess, ActionType act
 
 		//Important: There is no options for other roles like service, serviceUser and other!
 		if (resource != null && member != null) {
-			if (roles.contains(Role.VOADMIN)) {
+			if (roles.containsKey(Role.VOADMIN)) {
 				if (isAuthorized(sess, Role.VOADMIN, member)) return true;
 			}
-			if (roles.contains(Role.VOOBSERVER)) {
+			if (roles.containsKey(Role.VOOBSERVER)) {
 				if (isAuthorized(sess, Role.VOOBSERVER, member)) return true;
 			}
-			if (roles.contains(Role.FACILITYADMIN)) {
+			if (roles.containsKey(Role.FACILITYADMIN)) {
 				if (isAuthorized(sess, Role.FACILITYADMIN, resource)) return true;
 			}
-			if (roles.contains(Role.SELF)) {
-				if (isAuthorized(sess, Role.SELF, member)) return true;
+			if (roles.containsKey(Role.SELF)) {
+				if (roles.get(Role.SELF).contains(ActionType.READ_PUBLIC) || roles.get(Role.SELF).contains(ActionType.WRITE_PUBLIC)) return true;
+				if (roles.get(Role.SELF).contains(ActionType.READ) || roles.get(Role.SELF).contains(ActionType.WRITE)) {
+					if (isAuthorized(sess, Role.SELF, user)) return true;
```

Comment:
Shouldn't be here `member` instead of `user`? Here the user will be always null.